src/exactio/exactio_exanic.c

```diff
@@ -35,14 +35,22 @@ typedef enum {
 } exactio_exa_mod_t;
 
 
+static int set_exanic_params(exanic_t *exanic, char* device, int port_number,
+                             bool promisc, bool kernel_bypass);
 
-static void exa_destroy(eio_stream_t* this)
+void exa_destroy(eio_stream_t* this)
 {
     exa_priv_t* priv = IOSTREAM_GET_PRIVATE(this);
+
     if(priv->closed){
         return;
     }
 
+    if(set_exanic_params(priv->rx_nic, priv->rx_dev, priv->rx_port, 0, 0))
+    {
+        printf ("Unable to restore promisc and kernel bypass mode\n");
```

Comment:
Can you please use the ch_log functions for this? e.g. ch_log_error() or ch_log_fatal(). These should be used consistently through the codebase and will then end up in the correct logs for users.